Task-54516: Event list not displayed after changing timezone.

montassarhamadi opened the pull request:
Problem: if your agenda is synchronized with google calendar . when clicking on event .all events on the same day were displaying but when we change time zone we can't find those events .SO the problem was in the incorrect format user's timezone. that's why the response status of the callback was 400 (bad request ) invalid parameters.

Fix: in this fix, we formatted the time zone of the user as it should be (hh:mm) And of course with the sign that should be in front of hours.

rdenarie reviewed (changes requested):
Shouldnt we fix the getUserTimeZone function to have the correct format ?
In fact, you modify  a function used from other location ? How can you be sure that you do not make collateral impact ?

montassarhamadi commented:
>No romain because this function (getUserTimeZone) is not related to what I modified just I change the two parameters which used to filter all events in reality two those parameters should be in this format
-timeMin  =yyyy-mm-ddT00:00:00.000Z
-timeMax = yyyy-mm-ddT23:59:00.000Z
if you want to check just test this fix in locale .

rdenarie commented:
Your solution completly remove the timezone from the google request.
But in fact, the problem is that the function getTimezone is bugged when timezone is negative.
Please check this jsfiddle : https://jsfiddle.net/Lwz3p4m5/
firstly with european timezone, then with GMT-6 timezone
With GMT+1, the result of this function is  
![image](https://user-images.githubusercontent.com/807839/157018415-5a3d7f64-f2f9-4175-80dc-723c0f46f90f.png)

With GMT-6, the result is : 
![image](https://user-images.githubusercontent.com/807839/157018517-d58826de-61f1-444b-952b-6e2273af6a59.png)
but it must be 
-06:00 to be complient with the rfc format.

So, please fix the function getUserTimezone instead of removing it !

Be sure to test some exotic timezone like UTC-9h30 (polynesine francaise iles marquise)
Currently it display this 
![image](https://user-images.githubusercontent.com/807839/157019008-a67b9229-04d0-4aa6-80bb-7baa719f2b84.png)

instead of -09:30

montassarhamadi committed:
Task-54516: Event list not displayed after changing timezone.

Problem: if your agenda is synchronized with google calendar . when clicking on event .all events on the same day were displaying but when we change time zone we can't find those events .SO the problem was in the incorrect format user's timezone. that's why the response status of the callback was 400 (bad request ) invalid parameters.

Fix: in this fix, in this fix, we formatted the time zone of user as it should be (hh:mm) And of course with the sign that should be in front of hours.
